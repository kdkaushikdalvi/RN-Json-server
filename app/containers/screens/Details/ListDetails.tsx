import PropTypes from 'prop-types';
import React, {useState, useEffect} from 'react';
import {Text, View, Image, TouchableOpacity, ScrollView} from 'react-native';
import _Loader from '../../../components/Loading/_Loading';
import {getSongList} from '../../../service/api';
import styles from './styles';

interface IDetailsListProps {
  route: any;
}
interface IParams {
  id: string;
  title: string;
  artist: string;
  duration: string;
  image: string;
}

const defaultParams = {
  id: '',
  title: '',
  artist: '',
  duration: '',
  image: '',
};

export function ListDetails(props: IDetailsListProps): JSX.Element {
  const [loading, setLoading] = useState(true);
  const [user, setSong] = useState<IParams>(defaultParams);
  const {itemId} = props?.route?.params || 1; // 1 is test id
  useEffect(() => {
    const getSongs = async () => {
      const response = await getSongList(itemId);
      setSong(response.data);
    };
    getSongs();
    setTimeout(() => {
      setLoading(false);
    }, 1000);
  }, []);

  return (
    <ScrollView>
      <View style={styles.container}>
        <View style={styles.postContent}>
          <View style={styles.profile}>
            <Image style={styles.avatar} source={{uri: user.image}} />
          </View>
          <Text style={styles.postTitle}>Song Name : {user.title}</Text>
          <Text style={styles.postDescription}>
            Description : {user.artist}
          </Text>
          <Text style={styles.date}>Duration : {user.duration}</Text>
          <TouchableOpacity style={styles.shareButton}>
            <Text style={styles.shareButtonText}>Play</Text>
          </TouchableOpacity>
        </View>
      </View>
      {loading && <_Loader visible={true} />}
    </ScrollView>
  );
}
