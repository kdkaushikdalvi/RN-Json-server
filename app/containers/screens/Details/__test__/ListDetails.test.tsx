import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import {expect, it} from '@jest/globals';
import {ListDetails} from '../ListDetails';
import renderer from 'react-test-renderer';

describe('Some component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ListDetails />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('render ListDetails component ', async () => {
    const {getByText, getByTestId, debug} = render(<ListDetails />);
  });
});
