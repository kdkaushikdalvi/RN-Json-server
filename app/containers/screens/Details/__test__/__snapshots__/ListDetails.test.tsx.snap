// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some component renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "padding": 30,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#00BFFF",
                "borderRadius": 35,
                "borderWidth": 4,
                "display": "flex",
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "fontSize": 26,
              "fontWeight": "600",
            }
          }
        >
          Song Name : 
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginTop": 10,
            }
          }
        >
          Description : 
        </Text>
        <Text
          style={
            Object {
              "color": "#696969",
              "marginTop": 10,
            }
          }
        >
          Duration : 
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#7DE695",
              "borderRadius": 30,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginTop": 10,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Play
          </Text>
        </View>
      </View>
    </View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000077",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ActivityIndicator
          color="white"
          size="large"
        />
      </View>
    </Modal>
  </View>
</RCTScrollView>
`;
